redash/query_runner/rockset.py

```diff
@@ -56,7 +57,10 @@ def collection_columns(self, workspace, collection):
         return sorted(set([x["field"][0] for x in response["results"]]))
 
     def query(self, sql):
-        return self._request("queries", "POST", {"sql": {"query": sql}})
+        if (self.vi_id is not None and self.vi_id != ""):
+            return self._request("virtualinstances/{}/queries".format(self.vi_id), "POST", {"sql": {"query": sql}})
+        else:
+            return self._request("queries", "POST", {"sql": {"query": sql}})
```

Comment:
To reduce repetition, one alternative here could be:

```python
if self.vi_id is not None and self.vi_id != ""):
  query_path = f"virtualinstances/{self.vi_id}/queries"
else:
  query_path = "queries"
return self._request(query_path, "POST", {"sql": {"query": sql}})
```

More broadly, should we enforce elsewhere that `vi_id` should not ever be `""`? I think it would probably be better to have it either be a (real) string or `None`.

Reply:
I have changed this and tested, `make format` doesn't report any errors. I'll commit now.
I think the check for `vi_id` is OK as it is - I tested the connectivity with both code paths and it works well.